### PR TITLE
Feat/manager shell callback

### PR DESCRIPTION
**Pull request by frenautvh**
| Question         | Answer
| ---------------- | ---
| Branch?          | feat/manager-shell
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-7942
| License          | BSD 3-Clause

## Description

Refactor ovh-shell component to add callback / listener capabilities between shell client & shell

### CODE REVIEW

File: packages/components/ovh-shell/features/client/shell-client.feature [hunk 1]
```diff
@@ -6,3 +6,7 @@ Scenario: Plugin method invokation
   When I invoke a method from my plugin
   Then The invokation should be resolved with the method result
 
+Scenario: Plugin event listener
+  When My shell and shell client are configured with a direct message bus
```
Comment (zakariasahmane) [inline packages/components/ovh-shell/features/client/shell-client.feature:10]: `When` should be used when an action happens. This line is more related to configuration/prerequisites, so it should be `Given`. 

Also, to respect the Gherkin pattern, we should always have `Given, When, Then` and they are nearly the same as the AAA pattern ( Arrange, Act, Assert ). 

In this case, it should be more like : 

`Given My shell and shell client are configured with a direct message bus`
`And I add an event listener on the shell client`
`When The shell emits an event`
`Then The event emitted from the shell should be received by the client`

File: packages/components/ovh-shell/features/client/shell-client.feature [hunk 2]
```diff
@@ -6,3 +6,8 @@ Scenario: Plugin method invokation
   When I invoke a method from my plugin
   Then The invokation should be resolved with the method result
 
+Scenario: Plugin event listener
+  Given My shell and shell client are configured with a direct message bus
+  And I add an event listener on the shell client
+  When Then shell emits an event
```
Comment (Jisay) [inline packages/components/ovh-shell/features/client/shell-client.feature:12]:
```suggestion
  When Shell emits an event
```
?

File: packages/components/ovh-shell/src/client/api.ts [hunk 3]
```diff
@@ -1,19 +1,29 @@
...
   return {
     i18n: {
       getLocale: () =>
-        shell.invokePluginMethod({
+        shellClient.invokePluginMethod({
           plugin: 'i18n',
           method: 'getLocale',
         }),
+      onLocaleChange: (callback: CallableFunction) =>
+        shellClient.addEventListener('locale-change', callback),
```
Comment (Jisay) [inline packages/components/ovh-shell/src/client/api.ts:12]: Maybe it should be a good idea to prefix the event name with the name of the plugin?
Something like `i18n:locale-change`?
I searched for best practice and I found this article: https://gomakethings.com/custom-event-naming-conventions-in-vanilla-js/

**Review by cbourgois — APPROVED** [review]

**Review by Jisay — APPROVED** [review]

**Review by zakariasahmane — APPROVED** [review]
Nice !

**Pull request merged**